include/nlohmann/json.hpp

```diff
@@ -1394,6 +1417,24 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         return m_data.m_type;
     }
 
+#if DIAGNOSTIC_POSITIONS
+    constexpr size_t start_pos() const noexcept
+    {
+        return start_position;
+    }
+
+    constexpr size_t end_pos() const noexcept
+    {
+        return end_position;
+    }
+  private:
+    /// the start position of the value
+    size_t start_position = std::string::npos;
```

Comment:
I think these data members should be next to the other data members, down around 4265, probably after the existing ones so as to not unnecessarily change the order of data members.  We also need to make sure that the members are initialized in the constructors in same order, as compilers will warn if they're not in the same order.

Reply:
Made a judgement call on the location, thanks for providing the appropriate one. I’ll make sure to run the ci tools locally for the compiler checks